### PR TITLE
Add plugin: All Tasks Here

### PR DESCRIPTION
**Pull request by arkimedz-gh**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/arkimedz-gh/obsidian-all-tasks-here

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [X]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by liamcain** [comment]
- [await](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/main.ts#L83) this await isn't doing anything, you can remove it
- [this.app.workspace.getLeavesOfType(VIEW_TYPE_TODO).forEach((leaf) => leaf.detach());](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/main.ts#L89) <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>
- [props](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/main.ts#L34) why define these here? Why not move these functions into the View itself?
- [onClickEvent](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/src/components/TodoItemView.ts#L113) You should use [`Keymap.isModEvent`](https://github.com/obsidianmd/obsidian-api/blob/009a10f196df551256a5894ea60fdf9ddaab19a1/obsidian.d.ts#L1639) and use that for your "openFile" configuration. That way opening files will be consistent across the app.
- [const view = this.app.workspace.getActiveViewOfType(MarkdownView);](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/main.ts#L46) Are you trying to find the view associated with the current leaf here? You can just set the view state using openFile's second argument. You can provide a scroll offset or a cursor position directly
- [openFileInLeaf(this.app.workspace.getUnpinnedLeaf());](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/main.ts#L62) this is deprecated
- [/(-|\*) \[(\s|x)?\]\s(.*)/g;](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/src/models/TodoParser.ts#L14) Obsidian tasks support any character inside the [ ] brackets. So this needs to be more generic. It also needs to be case-insensitive.
- [containerEl.createEl('h2', { text: 'Obsidian TODO Settings' });](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/src/ui/SettingsTab.ts#L20) https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.
- [entry[2] === 'x'](https://github.com/arkimedz-gh/obsidian-all-tasks-here/blob/5cd739994bba443f54e07a49dbae0520fa9a27fe/src/models/TodoParser.ts#L22) this is forgetting about "X"
